### ISSUE TITLE
GROMACS - issue continuing a simulation

### ISSUE THREAD

**Issue by fadelm01** [Colvars/colvars#651]
hello everyone,

I'm using GROMACS 2023.3, colvars version 2023-12-05.
I want to continue a simulation where I'm using colvars to investigate a particular cholesterole molecule flopping around in a stratum corneum membrane - I'm doing my simulations on a cluster, so I'm splitting up my simulation into smaller jobs.

To continue the simulation from an older job, I want to use the md.colvars.state file (renaming it into md.colvars.state.dat), like this: [excerpt from my slurm input]
```
gmx mdrun \
    -s npt-prod-step1.tpr \
    -gpu_id $CUDA_VISIBLE_DEVICES \
    -maxh 72 \
    -colvars previous/my_config.colvars.dat \
    -colvars_restart ../part01/md.colvars.state.dat
```
The folder where I'm running the simulation contains:
- the slurm input
- the colvars configuration (in the subfolder "previous")
- the .tpr file
- the index file I need for my colvars configuration

As visible from the slurm input, the colvars.state.dat file is always in a different folder, so not in the same folder where I'm running the simulation.

I attached the files I used in this .zip:
[Archive.zip](https://github.com/Colvars/colvars/files/14164052/Archive.zip)

The job where I continue the simulation always crashes after six seconds, and always gives the same error: 

> colvars: Error: cannot open restart file/channel "md".

I've read the colvars 2023-12-05 manual (gromacs edition) but it doesn't indicate what I'm doing incorrectly, as it only refers to GROMACS 2024.

> " Beginning with GROMACS 2024, all information necessary to restart Colvars is included in the checkpoint “.cpt" file. No special provisions are therefore needed compared to a GROMACS simulation without Colvars enabled."

I would be incredibly thankful if someone could help me out with this.
Thank you very much and definitely let me know if I haven't given enough information on something.

**Comment by jhenin** [comment]
Hi, thanks for the report. We need more information to reproduce the issue.
Can you upload the Colvars configuration and a gro file please?

**Comment by fadelm01** [comment]
[gro-colvars.zip](https://github.com/Colvars/colvars/files/14166312/gro-colvars.zip)
Thank you for your answer, I uploaded the gro file and the colvars configuration into this .zip now.

**Comment by fadelm01** [comment]
my folder structure is like this:

production is the parent folder
it contains part01, part02-test (which is where I attempted to run the simulation).

Also, I'm pretty sure I just gave you my outdated confout.gro. This should be the correct one: 
[confout.gro.zip](https://github.com/Colvars/colvars/files/14166509/confout.gro.zip)

Sorry for the error, my brain is not braining because I've tried fixing this since saturday. 
I hope you will be able to reproduce my error now.

I've also attempted to give it the absolute path rather than the relative path (also from different folders that aren't part02-test), but it always gave the same error that "cannot open restart file/channel: md"

**Comment by fadelm01** [comment]
I've been fucking around with this for a while now and talked to my professor as well:

**both the .state and the .state.dat file need to exist**

Essentially, my error was that I just renamed the .state file to .state.dat (as gromacs requires you to), and didn't "keep" the .state file too.
I hope this could be clarified in either the error message you receive when you don't have a .state.dat file, or somewhere in the manual or something. 
(I think we could close this thread now?)

either way, thank you very much @jhenin :)

**Comment by jhenin** [comment]
This comes as a surprise to me - in my tests I did rename the file, not copy it. Anyway I'm glad that this is working for you. We will probably not push this further, since this interface is going to be deprecated by Gromacs 2024.

**Comment by HubLot** [comment]
Hi @fadelm01,

It looks like the bug you encountered was fixed in https://github.com/Colvars/colvars/pull/642.
It was merged on 8.12.2023 and your version is 5.12.2023.

So, if you want, you can patch manually your GROMACS installation with the master version of colvars and the bug should not arise.

**Comment by giacomofiorin** [comment]
@HubLot @fadelm01 In addition to or as an alternative to patching, how about using the pre-patched version of 2023?

https://github.com/Colvars/gromacs/releases/tag/v2023.4-colvars

I just updated that repo to include the latest patch release of the standard GROMACS, 2023.4.
